SQLite/Core/Connection.swift

```diff
@@ -25,8 +25,209 @@
 import Dispatch
 import CSQLite
 
+
+/// The mode in which a transaction acquires a lock.
+public enum TransactionMode : String {
+    
+    /// Defers locking the database till the first read/write executes.
+    case Deferred = "DEFERRED"
+    
+    /// Immediately acquires a reserved lock on the database.
+    case Immediate = "IMMEDIATE"
+    
+    /// Immediately acquires an exclusive lock on all databases.
+    case Exclusive = "EXCLUSIVE"
+    
+}
+
+
+/// Protocol to an SQLite connection
+public protocol Connection {
```

Comment:
I wonder if we can find another, happier medium. The reaction I had to the current naming direction is that it's a little harder to derive specific meaning on its own. I'm left wondering why only one of the types is prefixed with the abbreviation `DB` and what the significance of `DB` has over the other types. I also wonder if decorators like `DB` are slightly redundant in a SQLite library?

I actually prefer `ConnectionType` over `ConnectionProtocol`, but was trying to move in the general direction of Swift norms. I wonder if we could come up with a different protocol name that we like even better. Or maybe `DBConnection` could change to something else, like `ConcreteConnection`, though that descriptor still seems slightly confusing to me.

Any other ideas?